src/Token/SecureParser.php

```diff
@@ -0,0 +1,42 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Token;
+
+use Lcobucci\JWT\Decoder;
+use Lcobucci\JWT\SecureParser as SecureParserInterface;
+use Lcobucci\JWT\Token as TokenInterface;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\SignedWith;
+use Lcobucci\JWT\Validation\ValidAt;
+use Lcobucci\JWT\Validator;
+
+final class SecureParser implements SecureParserInterface
+{
+    private Parser $parser;
+    private Validator $validator;
+
+    public function __construct(Decoder $decoder, Validator $validator)
+    {
+        $this->parser    = new Parser($decoder);
```

Comment:
`Parser` dependency has been intentionally internalized: if and when it will be removed, `SecureParser` API should not be affected and we can just copy-paste its code here.